mungegithub/mungers/milestone-maintainer.go

```diff
@@ -283,6 +278,32 @@ func durationGreaterThanZero(name string, value time.Duration) error {
 	return nil
 }
 
+// parseMilestoneModes transforms a string containing milestones and
+// their modes to a map:
+//
+//     "v1.8=dev,v1.9=slush" -> map[string][string]{"v1.8": "dev", "v1.9": "slush"}
+func parseMilestoneModes(target string) (map[string]string, error) {
+	if len(target) == 0 {
+		return nil, fmt.Errorf("at least one milestone must be configured")
+	}
+
+	result := map[string]string{}
+	tokens := strings.Split(target, ",")
+	for _, token := range tokens {
+		parts := strings.Split(token, "=")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("expected format for each milestone is [milestone]=[mode], got '%v'", target)
+		}
+		milestone := strings.TrimSpace(parts[0])
+		mode := strings.TrimSpace(parts[1])
+		if !milestoneModes.Has(mode) {
+			return nil, fmt.Errorf("mode for milestone %v must be one of %v, but got '%v'", milestone, milestoneModes.List(), mode)
+		}
+		result[milestone] = mode
```

Comment:
Add a check that each milestone is only specified once.

Reply:
Done.